samtranslator/model/api/api_generator.py

```diff
@@ -147,7 +154,8 @@ def _construct_deployment(self, rest_api):
         deployment = ApiGatewayDeployment(self.logical_id + 'Deployment',
                                           attributes=self.passthrough_resource_attributes)
         deployment.RestApiId = rest_api.get_runtime_attr('rest_api_id')
-        deployment.StageName = 'Stage'
+        if not self.open_api_version:
+            deployment.StageName = 'Stage'
```

Comment:
I am so glad we are finally getting rid of this.

I proposed also getting rid of it if we detect that they're using OpenApi 3.x in their swagger (but don't have the OpenApi flag set), are we going ahead with that idea or not?

Reply:
Unfortunately there are cases where we don't read through the swagger document (definitionUri) so I don't think that is a consistent experience.

Reply:
Could you elaborate? We should be able to get rid of StageStage if they specify **any** value for `OpenApi` (i.e only if they don't specify a value do we keep it).

Reply:
Had an offline conversation about this, we will only fix stage stage if this flag is specified. If they don't have it specified and use oas 3 swagger in their api definitionbody it will still create an additional stage. This is because we want to remain backwards compatible and dont want to fix the stage stage bug without the customer opting in (with use of the OpenApiVersion flag)